packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -998,6 +1003,13 @@ abstract class PopupRoute<T> extends ModalRoute<T> {
 ///     // Do something
 ///   }
 ///
+///   @override
+///   Widget build(BuildContext context) => new Container();
+///
+/// }
+///
+/// class RouteAwareWidget extends StatefulWidget {
+///   State<RouteAwareWidget> createState() => new RouteAwareWidgetState();
 /// }
 /// ```
 class RouteObserver<T extends Route<dynamic>> extends NavigatorObserver {
```

Comment:
I don't understand the "if (route is T && previousRoute is T)" part of the API here.